test/mpi/manual/mpi_t/mpit_events_recvq.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+/* A simple test of the proposed MPI_T_ interface that assumes the
+ * presence of two events: "unexp_message_enqueued" and
+ * "unexp_message_dequeued".  Some information about these events
+ * is printed to stdout and then a few messages are sent/received to
+ * show that the call backs for these events are triggered.
+ *
+ */
+
+#include "mpi.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <assert.h>
+#include <math.h>
+#include <string.h>
+#include "mpitestconf.h"
+
+#if !defined(USE_STRICT_MPI) && defined(MPICH)
+
+char errMsg[1000];
+int errMsgLen;
+#define CHECKERR(errstr,err) if (err!=MPI_SUCCESS) { printf("ERROR: %s: MPI error code %i: ",errstr,err); MPI_Error_string(err, errMsg, &errMsgLen); errMsg[errMsgLen]=0; printf("%s\n", errMsg); /*usage(1);*/ }
+
+
+void uenq_cb_fn (MPI_T_event_instance event_instance, MPI_T_event_registration er, MPI_T_cb_safety cb_safety, void *user_data)
+{
+   
+   int rank;
+   int tag; 
+   MPI_T_event_read(event_instance, 0, &rank);
+   MPI_T_event_read(event_instance, 1, &tag);
+   printf("UENQ_CB_FN: rank: %d tag: %d \n", rank, tag);
+}
+
+void udeq_cb_fn (MPI_T_event_instance event_instance, MPI_T_event_registration er, MPI_T_cb_safety cb_safety, void *user_data)
+{
+   int rank;
+   int tag; 
+   MPI_T_event_read(event_instance, 0, &rank);
+   MPI_T_event_read(event_instance, 1, &tag);
+   printf("UDEQ_CB_FN: rank: %d tag: %d \n", rank, tag);
+}
+void recvq_free_cb (MPI_T_event_registration event_registration, MPI_T_cb_safety cb_safety, void *user_data)
+{
+    printf("FREE_CB_FN\n");
+}
+int main(int argc, char **argv)
```

Comment:
Let's see if we can move this test into the `test/mpi/impls/mpich/mpi_t` directory, which is part of our regular Jenkins testing. There will need to be modifications to the test to fit it into the MPICH test harness. I can explain in more detail.

Reply:
ok